### PR TITLE
Update to a version of Rubocop that is compatible with Ruby 2.4-3.1

### PR DESCRIPTION
**Pull request by petergoldstein**
This is an update to Rubocop that maintains compatibility with Ruby 2.4-3.1 using a common .rubocop_rspec_base.yml file across `rspec-core`, `rspec-expectations`, `rspec-mocks`, and `rspec-support`

Changes to this common base configuration file are minimal, limited to:

1. Adding missing Departments for Cops
2. Splitting/renaming Cop configurations where directed by Rubocop for pre-existing configurations.
3. Removing a Performance cop configuration as that is now packaged in a separate library.

Each repo now has a .rubocop_todo.yml file which can be worked through at the maintainers leisure.

Some updates to the existing .rubocop.yml files were made, mostly in removing duplication and updating some variables so Rubocop would run green.

I also updated cucumber to allow ruby-head to pass.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 inherit_from:
+  - .rubocop_todo.yml
   - .rubocop_rspec_base.yml
 
+
+
```
Comment (JonRowe) [inline .rubocop.yml:6]: No need for whitespace

File: .rubocop.yml [hunk 2]
```diff
@@ -9,20 +12,20 @@ Metrics/ClassLength:
 Metrics/CyclomaticComplexity:
   Max: 18
 
-Metrics/LineLength:
-  Max: 193
+Layout/LineLength:
+  Max: 325
```
Comment (JonRowe) [inline .rubocop.yml:16]: Whats going on with these increases?

File: .rubocop_rspec_base.yml [hunk 3]
```diff
@@ -4,76 +4,73 @@
 # This file contains defaults for RSpec projects. Individual projects
 # can customize by inheriting this file and overriding particular settings.
 
-AccessModifierIndentation:
+Layout/AccessModifierIndentation:
```
Comment (JonRowe) [inline .rubocop_rspec_base.yml:7]: We generally update this file using rspec-dev that can make multiple changes at once, I'd prefer to do that and then add the extra "version bump" commit to those prs by hand.

File: .rubocop.yml (hunk 1 above)
Comment (petergoldstein) [reply .rubocop.yml]: Fixed.

**Comment by petergoldstein** [comment]
I pulled out the Cucumber changes into a separate PR.

File: .rubocop_rspec_base.yml (hunk 3 above)
Comment (pirj) [reply .rubocop_rspec_base.yml:7]: Checked - .rubocop_rspec_base.yml is identical across PRs.
@petergoldstein Would you please send a PR to `rspec-dev` to update it there, too?

File: .rubocop.yml [hunk 4]
```diff
@@ -9,20 +10,27 @@ Metrics/ClassLength:
...
+  Max: 47
 
 Metrics/BlockLength:
   Max: 45
+  Exclude:
+    - spec/**/*
 
 Metrics/ModuleLength:
-  Max: 210
+  Max: 215
```
Comment (pirj) [inline .rubocop.yml:31]:
```
lib/rspec/mocks/message_expectation.rb:409:7: C: Metrics/ModuleLength: Module has too many lines. [220/215]
      module ImplementationDetails ...
```
:+1: to increase to 220

File: .rubocop_rspec_base.yml (hunk 3 above)
Comment (petergoldstein) [reply .rubocop_rspec_base.yml:7]: @pirj Done here - https://github.com/rspec/rspec-dev/pull/297

**Pull request merged**